### PR TITLE
Include entrypoint

### LINKED ISSUE #25: Include entrypoint for backend
- [ ] Starts the server
- [ ] Starts the database driver

### PR DESCRIPTION
**Pull request by kwannoel**
Closes #25 

Rebased off #26 (If this gets merged, 26 can be closed)

## Details

Starts the neo4j Driver before the initializing the express server.

This allows the server routers to use the driver object to forward requests to neo4j.

### CODE REVIEW

File: backend/src/server.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
-const express = require('express')
-const app = express()
-const port = process.env.PORT || 5000
+function runServer (driver) {
+  const express = require('express')
```
Comment (arbus) [inline backend/src/server.js:2]: Always `require` at the top level

**Comment by kwannoel** [comment]
> The following error occurs:
> 
> (node:3923) UnhandledPromiseRejectionWarning: TypeError: Bolt URL expected to be string but was: undefined
> at assertString (/home/daniel/HolmuskDB/node_modules/neo4j-driver/lib/internal/util.js:103:11)
> at Object.driver (/home/daniel/HolmuskDB/node_modules/neo4j-driver/lib/index.js:241:26)
> at runDriver (/home/daniel/HolmuskDB/backend/src/db-utils.js:4:24)
> at main (/home/daniel/HolmuskDB/backend/src/index.js:5:24)
> at Object. (/home/daniel/HolmuskDB/backend/src/index.js:9:1)
> at Module._compile (internal/modules/cjs/loader.js:1200:30)
> at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
> at Module.load (internal/modules/cjs/loader.js:1049:32)
> at Function.Module._load (internal/modules/cjs/loader.js:937:14)
> at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:3923) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
> (node:3923) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> 
> In addition, it seems that perhaps the runServer function doesn't actually run as console.log('Listening on Port ...') does not get printed out on the console nor does the server actually start as I cannot connect to the localhost.

@dzhou343 

Your environment variables are not loaded.

Run `source env.sh` before `npm start`

You can reference the [README](https://github.com/Holmusk/HolmuskDB/tree/master/backend#environment)

**Comment by kwannoel** [comment]
> Encountered new error after new env.sh loaded.
> 
> Error Dump:
> 
> ```
> (node:3477) UnhandledPromiseRejectionWarning: Neo4jError: Failed to connect to server. Please ensure that your database is listening on the correct host and port and that you have compatible encryption settings both on Neo4j server and driver. Note that the default encryption setting has changed in Neo4j 4.0. Caused by: connect ECONNREFUSED 127.0.0.1:7687
>     at newError (/home/daniel/HolmuskDB/node_modules/neo4j-driver/lib/error.js:75:10)
>     at NodeChannel._handleConnectionError (/home/daniel/HolmuskDB/node_modules/neo4j-driver/lib/internal/node/node-channel.js:229:41)
>     at Socket.emit (events.js:315:20)
>     at emitErrorNT (internal/streams/destroy.js:100:8)
>     at emitErrorCloseNT (internal/streams/destroy.js:68:3)
>     at processTicksAndRejections (internal/process/task_queues.js:84:21)
> (Use `node --trace-warnings ...` to show where the warning was created)
> (node:3477) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
> (node:3477) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```

@dzhou343 

You'll need to setup the docker instance for neo4j as well

Take a look at the README for instructions on how to do so

**Comment by kwannoel** [comment]
Thanks for the review @dzhou343 👍

**Pull request merged**